source/Octopus.Tentacle/Configuration/Instances/ConfigMapKeyValueStore.cs

```diff
@@ -0,0 +1,91 @@
+#if !NETFRAMEWORK
+using System;
+using System.Collections.Generic;
+using k8s;
+using k8s.Models;
+using Newtonsoft.Json;
+using Octopus.Diagnostics;
+
+namespace Octopus.Tentacle.Configuration.Instances
+{
+    class ConfigMapKeyValueStore : IWritableKeyValueStore, IAggregatableKeyValueStore
+    {
+        readonly string @namespace;
+        const string Name = "tentacle-config";
+        readonly k8s.Kubernetes client;
+
+        V1ConfigMap configMap;
+
+        IDictionary<string, string> ConfigMapData => configMap.Data ?? (configMap.Data = new Dictionary<string, string>());
+
+        public ConfigMapKeyValueStore(string @namespace, ISystemLog log)
+        {
+            this.@namespace = @namespace;
+            var kubeConfig = KubernetesClientConfiguration.InClusterConfig();
+            client = new k8s.Kubernetes(kubeConfig);
+            V1ConfigMap? config;
+            try
+            {
+                config = client.CoreV1.ReadNamespacedConfigMap(Name, @namespace);
+            }
+            catch
+            {
+                log.Verbose($"ConfigMap for Tentacle Configuration not found for namespace {@namespace}, creating new ConfigMap.");
+                config = null;
+            }
+
+            configMap = config ?? client.CoreV1.CreateNamespacedConfigMap(new V1ConfigMap { Metadata = new V1ObjectMeta { Name = Name, NamespaceProperty = @namespace }}, @namespace);
```

Comment:
To keep with our existing structure, this should be in a `KubernetesConfigMapService` and injected into this value store.

Reply:
Ah yep ok 👍